samples/place-autocomplete-data-session/index.ts

```diff
@@ -5,97 +5,146 @@
-    locationRestriction: { west: -122.44, north: 37.8, east: -122.39, south: 37.78 },
-    origin: { lat: 37.7893, lng: -122.4039 },
-    includedPrimaryTypes: ['restaurant'],
-    language: 'en-US',
-    region: 'us',
-};
-
-function init() {
-    titleElement = document.getElementById('title');
-    resultsContainerElement = document.getElementById('results');
-    inputElement = document.querySelector('input');
+    includedPrimaryTypes: [
+        'restaurant',
+        'cafe',
+        'museum',
+        'park',
+        'botanical_garden',
+    ],
+}
+
+async function initMap() {
+    await google.maps.importLibrary('maps');
+    innerMap = mapElement.innerMap;
+    innerMap.setOptions({
+        mapTypeControl: false,
+    });
+
+    // Update request center and bounds when the map bounds change.
+    google.maps.event.addListener(innerMap, 'bounds_changed', async () => {
+        request.locationRestriction = innerMap.getBounds();
+        request.origin = innerMap.getCenter();
+    });
+
     inputElement.addEventListener('input', makeAutocompleteRequest);
-    refreshToken(request);
 }
 
 async function makeAutocompleteRequest(inputEvent) {
+    // To avoid race conditions, store the request ID and compare after the request.
+    const requestId = ++newestRequestId
+
+    const { AutocompleteSuggestion } = (await google.maps.importLibrary(
+        'places'
+    )) as google.maps.PlacesLibrary;
+
     // Reset elements and exit if an empty string is received.
-    if (inputEvent.target.value == '') {
-        titleElement.innerText = '';
+    if (!inputEvent.target ||
+        inputEvent.target.value === '') {
+        titleElement.textContent = '';
         resultsContainerElement.replaceChildren();
         return;
     }
 
     // Add the latest char sequence to the request.
-    request.input = inputEvent.target.value;
-
-    // To avoid race conditions, store the request ID and compare after the request.
-    const requestId = ++newestRequestId;
+    request.input = (inputEvent.target as HTMLInputElement).value;
 
     // Fetch autocomplete suggestions and show them in a list.
-    // @ts-ignore
-    const { suggestions } = await google.maps.places.AutocompleteSuggestion.fetchAutocompleteSuggestions(request);
+    const { suggestions } =
+        await AutocompleteSuggestion.fetchAutocompleteSuggestions(request);
 
     // If the request has been superseded by a newer request, do not render the output.
     if (requestId !== newestRequestId) return;
 
-    titleElement.innerText = `Query predictions for "${request.input}"`;
+    titleElement.innerText = `Place predictions for "${request.input}"`;
 
     // Clear the list first.
     resultsContainerElement.replaceChildren();
 
     for (const suggestion of suggestions) {
         const placePrediction = suggestion.placePrediction;
 
+        if (!placePrediction) {
+            continue
+        }
+
         // Create a link for the place, add an event handler to fetch the place.
```

Comment:
```suggestion
// Create a button for the place, add an event handler to fetch the place.
```

I would just call it a button for simplicity